### PR TITLE
Modify the `build.sh` script to work with podman

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
In order to be able to build images directly in a wotkspace container, add `BUILDER` variable to be able to execute the `build.sh` script not only with docker but with podman and buildah.

depends on https://github.com/eclipse-che/che-server/pull/573
### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22243

### How to test this PR?

1. Start a workspace from the `che-server` repository
2. Git checkout to the `che-22243` brach
3. In the workspace terminal run `cd dockerfiles` and `.build.sh` commands

See: a new image has been built.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: dockerfiles/postgres/Dockerfile
Comment (ibuziuk) [inline dockerfiles/postgres/Dockerfile:19]: is this comment expected?

Comment (ibuziuk) [reply dockerfiles/postgres/Dockerfile:19]: in general wondering if we still need dockerfile for postgress since it was removed in 7.62.0 - https://che.eclipseprojects.io/2023/03/20/@ilya.buziuk-decommissioning-postgresql-database.html

Comment (vinokurig) [reply dockerfiles/postgres/Dockerfile:19]: This is a temporary solution, it is going to be removed by https://github.com/eclipse-che/che-server/pull/468

**Comment by tolusha** [comment]
Please, rebase on main to have tests passed.

File: dockerfiles/postgres/Dockerfile
Comment (ibuziuk) [reply dockerfiles/postgres/Dockerfile:19]: I'm still not clear with this change as part of podman switch.
Do we really need to update postrgres Dockerfile as part of this PR? 

@SDawley @nickboldt is #468 still actual which was opened in March

**Comment by ibuziuk** [comment]
@nickboldt @SDawley please, review
need your approval before merging + clarity with #468

File: dockerfiles/postgres/Dockerfile
Comment (vinokurig) [reply dockerfiles/postgres/Dockerfile:19]: @ibuziuk @nickboldt @SDawley I closed the https://github.com/eclipse-che/che-server/pull/468 in favor of https://github.com/eclipse-che/che-server/pull/573. This is the same PR but I merged the main branch into it, since the original PR was located in the personal fork.

File: devfile.yaml
Comment (nickboldt) [inline devfile.yaml:22]:
```suggestion
        mvn clean install -V -e -Pfast -DskipTests -Dskip-validate-sources -Denforcer.skip=true
```

Because the enforcer/validater tends to fail because of not updated license headers, this is a safer solution, and how we buildit downstream.

File: .github/workflows/next-build.yml
Comment (nickboldt) [inline .github/workflows/next-build.yml:52]:
```suggestion
```

File: .github/workflows/next-build.yml
Comment (nickboldt) [inline .github/workflows/next-build.yml:53]:
```suggestion
```

File: .github/workflows/next-build.yml
Comment (nickboldt) [inline .github/workflows/next-build.yml:55]:
```suggestion
```

File: .github/workflows/next-build.yml
Comment (nickboldt) [inline .github/workflows/next-build.yml:56]:
```suggestion
```

File: .github/workflows/next-build.yml
Comment (nickboldt) [inline .github/workflows/next-build.yml:58]:
```suggestion
```

File: .github/workflows/next-build.yml
Comment (nickboldt) [inline .github/workflows/next-build.yml:59]:
```suggestion
```

File: dockerfiles/build.include
Comment (nickboldt) [inline dockerfiles/build.include:171]: If you're adding support for `buildah` then you need to add the `BUILD_COMMAND` option here, and make sure it's `buildah bud` or `podman build` or `docker build`.

File: dockerfiles/build.include
Comment (nickboldt) [inline dockerfiles/build.include:107]:
```suggestion
  BUILD_COMAMAND="build"
  if [ -z $BUILDER ]; then
```

File: dockerfiles/build.include
Comment (nickboldt) [inline dockerfiles/build.include:171]:
```suggestion
  cd "${DIR}" && "${BUILDER}" "${BUILD_COMMAND}" -f ${DIR}/.Dockerfile -t ${IMAGE_NAME} ${BUILD_ARGS} .
```

Comment (vinokurig) [reply dockerfiles/build.include]: done

**Comment by vinokurig** [comment]
@ibuziuk @nickboldt @SDawley Please take a look

File: .github/workflows/next-build.yml
Comment (vinokurig) [inline .github/workflows/next-build.yml:54]:
```suggestion
```

**Comment by vinokurig** [comment]
@nickboldt  @ibuziuk Could you please review?

**Review by ibuziuk — APPROVED** [review]

**Pull request merged**
